app/src/main/java/zapsolutions/zap/walletManagement/WalletDetailsActivity.java

```diff
@@ -0,0 +1,208 @@
+            TextView tvHost = findViewById(R.id.host);
+            tvHost.setText(getWalletConfig().getHost());
+            TextView tvPortLabel = findViewById(R.id.portLabel);
+            tvPortLabel.setText(getResources().getString(R.string.port) + ":");
+            TextView tvPort = findViewById(R.id.port);
+            tvPort.setText(String.valueOf(getWalletConfig().getPort()));
+            TextView tvMacaroonLabel = findViewById(R.id.macaroonLabel);
+            tvMacaroonLabel.setText(getResources().getString(R.string.macaroon) + ":");
+            TextView tvMacaroon = findViewById(R.id.macaroon);
+            tvMacaroon.setText(getWalletConfig().getMacaroon());
+            TextView tvCertificateLabel = findViewById(R.id.certLabel);
+            tvCertificateLabel.setText(getResources().getString(R.string.certificate) + ":");
+            TextView tvCertificate = findViewById(R.id.cert);
+            tvCertificate.setText(getWalletConfig().getCert());
+
+            Button changeBtn = findViewById(R.id.buttonChangeConnection);
+            changeBtn.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Intent intent = new Intent(WalletDetailsActivity.this, ConnectRemoteNodeActivity.class);
+                    intent.putExtra(ManageWalletsActivity.WALLET_ID, mId);
+                    startActivity(intent);
+                }
+            });
+        }
+
+        Button switchBtn = findViewById(R.id.buttonActivate);
+        switchBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                PrefsUtil.edit().putString(PrefsUtil.CURRENT_WALLET_CONFIG, mId).commit();
+
+                // Do not ask for pin again...
+                TimeOutUtil.getInstance().restartTimer();
+
+                Wallet.getInstance().reset();
+
+                if (!getWalletConfig().isLocal()) {
+                    // ToDo: Stop local LND if we switch to a remote wallet (right now it stops, but then it crashes when switching back to Local LND
+                    openHome();
+                } else {
+                    openHome();
+                }
+            }
+        });
+
+        Button renameBtn = findViewById(R.id.buttonRename);
+        renameBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                showWalletNameInput();
+            }
+        });
+
+        Button deleteBtn = findViewById(R.id.buttonDelete);
+        deleteBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (PrefsUtil.getCurrentWalletConfig().equals(mId)) {
+                    Toast.makeText(WalletDetailsActivity.this, "You cannot delete the currently active wallet.", Toast.LENGTH_LONG).show();
```

Comment:
I though maybe we allow it, if its the only wallet in the app? To let the user get a clean state if he wants to delete them.

Reply:
It is now always possible to delete the currently active wallet. If you will delete it and there are other wallets, then it automatically switches to the first in the array. And if no wallets are left, you will be back in clean empty state.